File: 0217-contains-duplicate/0217-contains-duplicate.cpp
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {

        unordered_set<int> seen;

        for (int x : nums) {
            if (seen.find(x) != seen.end()) {
                return true;
            }

            seen.insert(x);
        }

        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {

        HashSet<Integer> seen = new HashSet<>();

        for (int x : nums) {
            if (seen.contains(x)) {
                return true;
            }

            seen.add(x);
        }

        return false;
    }
}